src/ShapeFile.php

```diff
@@ -326,8 +334,16 @@ public function getIndexFromDBFData(string $field, $value): int
      */
     private function loadDBFHeader(): array
     {
-        $DBFFile = fopen($this->getFilename('.dbf'), 'r');
+        if (! self::supportsDbase()) {
```

Comment:
I moved this block here so no matter where `loadDBFHeader` is called from support for `dbase` will always be checked.